avocado_vt/plugins/vt_joblock.py

```diff
@@ -0,0 +1,119 @@
+import errno
+import logging
+import os
+import re
+import random
+import string
+import sys
+
+from avocado.core import exit_codes
+from avocado.core.settings import settings
+from avocado.plugins.base import JobPre, JobPost
+
+from ..test import VirtTest
+
+
+class VTJobLock(JobPre, JobPost):
+
+    name = 'vt-joblock'
+    description = 'Avocado-VT Job Lock/Unlock'
+
+    def __init__(self):
+        self.log = logging.getLogger("avocado.app")
+        self.has_vt_test = False
+        self.lock_dir = settings.get_value(section="plugins.vtjoblock",
+                                           key="dir",
+                                           key_type=str,
+                                           default='/tmp')
+        self.show_lock = settings.get_value(section="plugins.vtjoblock",
+                                            key="show_lock_location",
+                                            key_type=bool,
+                                            default=False)
+        self.lock_file = None
+
+    def _abort(self, message, job):
+        """
+        Aborts the Job by exiting Avocado, adding a failure to the job status
+        """
+        self.log.error(message)
+        sys.exit(exit_codes.AVOCADO_JOB_FAIL | job.exitcode)
+
+    def _set_lock(self, job):
+        if not os.path.isdir(self.lock_dir):
+            msg = ('VT Job lock directory "%s" does not exist... '
+                   'exiting...' % self.lock_dir)
+            self._abort(msg, job)
+
+        pattern = 'avocado-vt-joblock-%(jobid)s-%(uid)s-%(random)s.pid'
+        # the job unique id is already random, but, let's add yet another one
```

Comment:
Job unique ID is __unique__. I means yes, theoretically if one user creates 2 identical jobs at exactly the same time with 2 different output directories, then it could produce the same jobid+uid. But I seriously doubt this would happen, so maybe we don't need the yet another random, what do you think?

(if the user uses the same results dir, than jobid is guaranteed to be unique; different users have different uid)